src/edu/csus/ecs/pc2/clics/API202306/CLICSScoreboardRow.java

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 1989-2024 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.clics.API202306;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import edu.csus.ecs.pc2.core.Utilities;
+import edu.csus.ecs.pc2.core.standings.ProblemSummaryInfo;
+import edu.csus.ecs.pc2.core.standings.TeamStanding;
+
+/**
+ * CLICS Scoreboard row
+ * Contains information about a single entry in the scoreboard
+ * 
+ * @author John Buck
+ *
+ */
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CLICSScoreboardRow {
+
+    @JsonProperty
+    private int rank;
+
+    @JsonProperty
+    private String team_id;
+
+    @JsonProperty
+    private CLICSScore score;
+    
+    @JsonProperty
+    private CLICSProblemScore [] problems;
+
+    /**
+     * Fill in API scoreboard row information properties (for scoreboard endpoint)
+     * 
+     * @param probEleToShort hashmap for mapping problem elementid to shortname
+     * @param teamStanding xml representation of the standings for a team
+     */
+    public CLICSScoreboardRow(HashMap<String, String> probEleToShort, TeamStanding teamStanding) {
+        team_id = teamStanding.getTeamId();
+        rank = Utilities.nullSafeToInt(teamStanding.getRank(), 0);
```

Comment:
Does "0" really represent a good default value for a rank?   (I could see "1", or "max_num_teams_on_scoreboard". But we _never_ assign "0" as a contest rank...)

Reply:
See my comments in CLICSScore.  If teamStanding.getRank() returns null, we have much more serious issues than returning 0 on the EF.  Since teamStanding comes from deserializing XML that PC2 created in DSA.  No changes made.

If you feel strongly about this, maybe file an issue with the suggested solution after the PR is merged.  This type of thing occurs in other places in the code (that's where I got it from).  So, if it gets changed here, it would have to be changed uniformly throughout, which I'm not going to do for this PR.

Reply:
I don't feel that strongly about it, and you're right that such a change would have to be propagated over a wide range.  Not something we want to do right now.